### ISSUE TITLE
A deamon for running application in the background

### ISSUE THREAD

**Issue by branja6** [Wunderfitz/harbour-fernschreiber#21]
Please, allow the application to run in the background so that we could receive notifications even if we close it.

**Comment by Bsoupe** [comment]
This should be adjustable! (I prefer to such a demon)!

**Comment by GNUuser** [comment]
Daemon icon is already. : ) just in case...
![daemon](https://user-images.githubusercontent.com/45316884/97119910-cb5d1c00-16e9-11eb-89b7-68b5cc972c16.png)

**Comment by tanghus** [comment]
Chiming in for a daemon 👍 Currently I have Depecher running for that, but sadly it's not that stable with notifications, plus it would make more sense to use the right app for it 😊

**Comment by Wunderfitz** [comment]
As we've achieved Harbour compatibility recently, a real daemon would only be introduced if somebody builds a proposal that still allows Fernschreiber to be built and packaged without it. As I personally will focus on other features and bug fixes, contributions are really welcome for that.
As a side note, I did some experiments to implement a Harbour-compliant daemon-like feature set (autostart the app and only send it to background if it's closed - both to be set optionally and independently from each other) in the branch https://github.com/Wunderfitz/harbour-fernschreiber/tree/stay-awake. Development is currently on hold here, but might be started again at a later time. Feedback on that approach is also appreciated!

**Comment by GNUuser** [comment]
Please  d a e m o n  feature.

**Comment by monich** [comment]
Out of curiosity, what's supposed to be the difference between daemon and the app swiped to the background?

**Comment by GNUuser** [comment]
Hidden application running, real notifications.

**Comment by monich** [comment]
> Hidden apps running, real notifications.

All that you also have when cover is visible. Is it the cover that you don't like?

**Comment by GNUuser** [comment]
> All that you also have when cover is visible. Is it the cover that you don't like?
I already know that with cover i have service running, but i feeling more comfortable with FernschreiberD.
Thank you @monich for the tips.

**Comment by ihababo** [comment]
@Wunderfitz 
As I'm no developer... Wouldn't it be possible to code the daemon as a standalone package/rpm that can be intsalled separately (and optionally)?
It would then be available on Openrepos and Fernschreiber remains compatible with Harbour.
Thats the only thing keeping me from using this great app unfortunately.

**Comment by Wunderfitz** [comment]
It might be possible, but someone simply needs to develop it. I'm open to contributions, but as I'm 100% satisfied with keeping the app in the background, there won't be a solution from me any time soon.

**Comment by ihababo** [comment]
Undersood. Thank you for your great work anyway.

**Comment by henning-schild** [comment]
I recently discovered fernschreiber but also will move back to Depecher, or install both just for the notifications.

While i could get used to keeping the app open and in the background, i find it hard to getting used to starting the app after every reboot ... the window where i missed messages since switching to fernschreiber. So maybe an auto-start could be a compromise and a daemon not actually needed.

**Comment by jgibbon** [comment]
@henning-schild that's a good point, you can try https://openrepos.net/content/blacksheepdev/takeoff for that, which is an auto start manager. I haven't tried that one, but apps by @black-sheep-dev tend to be pretty good.

**Comment by henning-schild** [comment]
@jgibbon good point, i might try that one. But installing anything from openrepos to add a missing feature just feels wrong, in that case i might as well install Depecher and be done with it.

I think the idea is that one can get a telegram client without the need to trust openrepos and toggling the "untrusted apps" config and without having to use an android client. The blocker seems to be that a "daemon app" seems to be hard to get into the harbour or that @Wunderfitz does not see i priority ... or probably a mix of both.

My story is that i recently had to reinstall my sailfish device and simply discovered fernschreiber as a possible alternative to depecher, at which point i was ready to go "trusted apps only". Turns out that did not really work ... with "notification issues" in telegram playing a central role to having to trust openrepos after all.

**Comment by jgibbon** [comment]
Seeing it as a missing feature is "just" a difference in perspective – having a small, dedicated tool managing auto starts can add a lot more value than relying on every program someone wants to run on system start to implement the same thing over and over again. (Basically the same reason I did a sleep timer without media player.) 
While it is technically possible with workarounds configured at runtime, autostarting daemons are not really something Jolla Store applications are "supposed to do".

The pros and cons of using "untrusted apps" are a valid concern, but out of scope for this issue IMHO. For the sake of this specific "workaround": If you wanted, you could look into [building takeoff yourself](https://github.com/black-sheep-dev/harbour-takeoff) .

A bit more on-topic: 
I considered building a daemon in the past, but it is a lot of work. As in 'rip out all of Fernschreibers insides, build a dbus wrapper for everything and then manage the daemon itself' – quite possibly slowing down and breaking a lot of things in the (pun not intended) process. So I don't think I'll tackle that, as well.

**Comment by carlosgonz0** [comment]
Please please please please consider implement a daemon.

**Comment by Wunderfitz** [comment]
Well, I have to say that things haven't changed that much since the last message. I'm open to contributions, but I definitely don't have the time or motivation myself. Sorry for that!
